### DISCUSSION TITLE
Problema di Connessione MQTT con Shelly 1

### DISCUSSION THREAD

**Discussion by BlockRockAdmin** [indomus/forum#2840] [Hardware & Firmware]
Ciao a tutti,

Sto riscontrando delle difficoltà nel connettere il mio interruttore Shelly 1 al protocollo MQTT e cerco aiuto o suggerimenti per risolvere questo problema.

Ho configurato il mio dispositivo Shelly 1 per connettersi al mio broker MQTT, ma sembra che non riesca a stabilire una connessione. Ho seguito tutte le istruzioni standard per la configurazione MQTT, ma senza successo.

    Modello Dispositivo: Shelly 1
    Broker MQTT: Mosquitto

Qualsiasi consiglio o suggerimento sarebbe molto apprezzato. Sono particolarmente interessato a sapere se qualcuno ha avuto esperienze simili con Shelly 1 e MQTT e come ha risolto il problema.

Grazie in anticipo per il vostro aiuto!

**Comment by snakuzzo** [comment]
Perdonami, ma con le informazioni che hai dato uno come fa ad aiutarti ? Va "a sensazione" ?
Dovresti dare qualche evidenza in più...come lo screenshot (non la foto) della configurazione MQTT dello Shelly ed il log del broker MQTT.

**Reply by BlockRockAdmin** [reply]
Ciao Diego @snakuzzo, sì, attualmente sto lavorando su un setup di test per capire quanto sia facilmente integrabile lo Shelly 1 con altri sistemi. 

Configurazione MQTT dello Shelly 1:
•	Ecco uno screenshot delle impostazioni MQTT sul mio Shelly 1:

<img width="338" alt="MQTT Shelly" src="https://github.com/indomus/forum/assets/4179048/925b7e69-8851-48e5-839d-e21bf8ade78f">

•	Indirizzo IP del Broker MQTT: 192.168.1.101
•	Indirizzo IP del Relè: 192.168.1.140
•	Porta: 1883
Log del Broker MQTT (Mosquitto):
•	Ecco le parti rilevanti del log di Mosquitto che mostrano tentativi di connessione dal mio Shelly 1 o eventuali errori:
 
Mentre in localhost riesco a visualizzare i messaggi pubblicati, come riportato dallo screen:

<img width="365" alt="cmd broker MQTT" src="https://github.com/indomus/forum/assets/4179048/82646171-3bb1-4159-9805-6f5291ae41d2">

Vorrei capire se lo Shelly 1 è completamente compatibile con il protocollo MQTT, in modo da poter testare la comunicazione attraverso le linee di comando. Leggendo informazioni online, ho notato che potrebbe essere necessaria anche l'abilitazione dell'RPC, una funzionalità che non sembrerebbe presente in questa versione del dispositivo. Se fosse possibile attivare questo processo, mi piacerebbe implementare un sistema domotico personalizzato nella mia abitazione utilizzando Python o PHP.

**Reply by snakuzzo** [reply]
Qualsiasi shelly è pienamente compatibile con protocollo MQTT. Il discorso RPC innanzi tutto è da fare solo con i device di gen 2 (i plus per intenderci) quindi non riguarda te, poi tutti gli shelly, senza alcuna eccezione, si interfacciano perfettamente con qualsiasi broker MQTT, che sia mosquitto o altro.

Dagli screenshot non si evince nulla:
Hai fatto un ping dal rpi al broker (che immagino sia lo stesso host) che non indica il funzionamento di nulla...al massimo avresti dovuto fare una prova con telnet (`telnet ipmosquitto 1883`) o una network connect (`nc -zvw2 ipmosquitto 1883`) da un altro client sulla stessa rete...e non da se stesso.

Quanto ai log, dagli screenshot non si evince nulla. Non serve lo stato del service, ma il log applicativo...che viene scritto nel path impostato nel file di configurazione di mosquitto (`mosquitto.conf`)
